### PR TITLE
ca_controller: Remove duplicate api call in updateAutoscaler

### PR DESCRIPTION
**Pull request by michaelgugino**
This commit removes extra call to api to get the existing
deployment.  This deployment is already retrieved in
Reconcile() method.

This commit also changes UpdateAutoscaler to private
updateAutoscaler.

### CODE REVIEW

File: pkg/controller/clusterautoscaler/clusterautoscaler_controller.go [hunk 1]
```diff
@@ -153,7 +153,7 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		return reconcile.Result{}, nil
 	}
 
-	if err := r.UpdateAutoscaler(ca); err != nil {
+	if err := r.updateAutoscaler(ca, dep); err != nil {
```
Comment (ingvagabund) [inline pkg/controller/clusterautoscaler/clusterautoscaler_controller.go]: Both `r.GetAutoscaler` are invoked pretty close to each other so the chance of changing the deployment between the calls is quite small. And, if that happens and `client.Update` complains about `the object has been modified; please apply your changes to the latest version and try again`, the deployment gets reconciled in the next loop.

File: pkg/controller/clusterautoscaler/clusterautoscaler_controller.go [hunk 2]
```diff
@@ -225,13 +225,9 @@ func (r *Reconciler) CreateAutoscaler(ca *autoscalingv1alpha1.ClusterAutoscaler)
...
 
-// UpdateAutoscaler will retrieve the deployment for the given ClusterAutoscaler
+// updateAutoscaler compares the deployment for the given ClusterAutoscaler
 // custom resource instance and update it to match the expected spec if needed.
-func (r *Reconciler) UpdateAutoscaler(ca *autoscalingv1alpha1.ClusterAutoscaler) error {
-	existingDeployment, err := r.GetAutoscaler(ca)
-	if err != nil {
-		return err
-	}
+func (r *Reconciler) updateAutoscaler(ca *autoscalingv1alpha1.ClusterAutoscaler, existingDeployment *appsv1.Deployment) error {
```
Comment (ingvagabund) [inline pkg/controller/clusterautoscaler/clusterautoscaler_controller.go]: +1 for making it public

File: pkg/controller/clusterautoscaler/clusterautoscaler_controller.go (hunk 1 above)
Comment (michaelgugino) [reply pkg/controller/clusterautoscaler/clusterautoscaler_controller.go]: Actually, they are invoked after checks of the err with no other operations in between.  If line 141 returns a deployment, the very next action (other than two boolean statements which resolve false) is to call GetAutoscaler again!

File: pkg/controller/clusterautoscaler/clusterautoscaler_controller.go (hunk 2 above)
Comment (michaelgugino) [reply pkg/controller/clusterautoscaler/clusterautoscaler_controller.go]: *private.

File: pkg/controller/clusterautoscaler/clusterautoscaler_controller.go (hunk 1 above)
Comment (bison) [reply pkg/controller/clusterautoscaler/clusterautoscaler_controller.go]: I get the reasoning here, but I don't think this is performance critical and we have caching. This makes calling the update harder and more error prone if it were to be called elsewhere. I think the original focus was on the API being friendly.

Comment (bison) [reply pkg/controller/clusterautoscaler/clusterautoscaler_controller.go]: So, if we're going to make `updateAutoscaler()` private and require passing in the deployment to compare, my vote would probably be to just inline it in `Reconcile()` instead.

Comment (michaelgugino) [reply pkg/controller/clusterautoscaler/clusterautoscaler_controller.go]: @bison I disagree.  I'm not sure what context we would potentially call updateAutoscaler() if we didn't already have a deployment that we needed to update.

I don't agree with inlining, I like it as-is for now.

**Pull request closed without merging**